cmd/processor/app/processor.go

```diff
@@ -492,3 +501,11 @@ func (p *Processor) detectPlatformKind() (string, error) {
 
 	return "local", nil
 }
+
+func (p *Processor) startTimeoutWatcher(eventTimeout time.Duration) {
+	if eventTimeout == 0 {
+		eventTimeout = clock.DefaultResolution
+	}
+	p.logger.InfoWith("Starting event timeout watcher", "timeout", eventTimeout.String())
+	timeout.NewEventTimeoutWatcher(p.logger, eventTimeout, p)
```

Comment:
> 1. New must always return error

Even if it can't fail? Is this for future proofing?

> 2. Save the instance

OK

> 3. I don't think clock should have a default resolution, rather function should have a default timeout

I don't understand. What do does what "function should have a default timeout" mean.
The reason we need a default clock is since triggers don't have access to the processor or the whole configuration. I'd like to make the API simple that they just grab the current time from the default clock (very much like the design in `http.Get` which calles `http.DefaultClient.Get`).

Reply:
> Even if it can't fail? Is this for future proofing?
Yes, especially creation of objects

Reply:
> The reason we need a default clock
I don't think 10 seconds is a "sane" default. That is to say, it seems to me that the clock package (which is generic) should force the user to think about what the resolution is. Without the user supplying a resolution, it's impossible to guess what the behavior will be when you use the package

As for the processor, if the event timeout is 0 - what does this mean?
That there's no timeout? If so, why do we even need to trigger a clock thingie in the background?